src/services/data-provider.js

```diff
@@ -74,18 +82,22 @@ export default {
     return httpClient(endpoint.url, {
       method: "POST",
       body: JSON.stringify(params.data),
-    }).then(({ json }) => ({
-      data: endpoint.getData(json.data),
-    }));
+    })
+      .then(({ json }) => ({
+        data: endpoint.getData(json.data),
+      }))
+      .catch(Promise.reject);
   },
 
   delete: (resource, params) => {
```

Comment:
some kinda error here, haven't completed testing

Reply:
The httpClient was not being returned. I have now fixed this.